R/similarProducts.R

```diff
@@ -72,10 +73,11 @@ keepOnePerGroup <- function(dt, groups) {
   if(is.null(groups)) return(dt)
 
   # Append group data to affinity table
-  groups.table <- data.table(sku = names(groups), group = groups, key= "sku")
+  groups.table <- data.table(sku = names(groups), group = groups, key = "sku")
   dt <- dt[groups.table, nomatch = 0]
 
   # Get the best performing sku per group
   # http://stackoverflow.com/questions/16573995/subset-by-group-with-data-table
-  dt <- dt[dt[, .I[sim == max(sim)], by = group]$V1]
+  by.cols <- setdiff(colnames(dt), c("sku", "sim"))
```

Comment:
here we usually would have 4 columns: `visitor.id`, `sku`, `group`, `sim`. 

So I guess alternative will be setting `by.cols` to `c("visitor.id", "group")`